wit/types.wit

```diff
@@ -1,44 +1,46 @@
-interface messaging-types {
+interface types {
     /// A connection to a message-exchange service (e.g., buffer, broker, etc.).
     resource client {
         connect: static func(name: string) -> result<client, error>;
     }
     
-    /// TODO(danbugs): This should be eventually extracted as an underlying type for other wasi-cloud-core interfaces.
-    resource error {
-        trace: static func() -> string;    
+    /// Errors that can occur when using the messaging interface.
+    variant error {
+        /// The requested option is not authorized. This could be a topic it doesn't have
+        /// permission to subscribe to, or a permission it doesn't have to perform a specific
+        /// action. This error is mainly used when calling `set-subscriptions` on a guest.
```

Comment:
```suggestion
        /// The requested option is not authorized. 
        /// This could be because a topic was not permissive to being
        /// subscribed to, or because permissions were lacking to
        /// perform some other specific action. This error is mainly used 
        /// when calling `set-subscriptions` on a guest
```
~nit